### DISCUSSION TITLE
在单一向量搜索时，如何配置按百分比过滤，比如查出相似度80%的topK条数据

### DISCUSSION THREAD

**Discussion by 262146150** [milvus-io/milvus#33855] [Q&A and General discussion]
在单一向量搜索时，如何配置按百分比过滤，比如查出相似度80%的topK条数据。
产品中，用户可以设置 topK 和 相似度，因为数据库用L2类型过滤，返回的距离用户肯定是不理解的，
所以最简单的设置办法就是百分比，不能设置浮点型的过滤条件，能否做到呢

**Comment by yhmo** [comment]
L2距离能获知哪条数据最为相似，但距离和相似度并不是线性关系，个人觉得没法精确找到80%相似的东西。
所谓“相似”，本身就是个模糊的概念，如果让你去评论两个人有百分之几相似，恐怕也很难得出一个具体的百分比。刘德华和张学友在有些人眼里可能是80%相似，在有些人眼里可能只有20%相似。

**Reply by 262146150** [reply]
感谢回复，想了一个折中的方案，默认过滤出 1到0之间的距离，然后在搜索结果中，计算每一条结果( 1 / (1 + $distance) )*100 的值，选出>=设置的百分比的值，不知道是否合理

**Reply by yhmo** [reply]
其实跟产生向量的模型也相关，不同的模型你这算法得出的数值都不一样。在不换模型的条件下，把这个式子调节到满意就好。可以写成 (x/(x+distance)*100来调那个x的值吧，我也是瞎蒙的不一定合理。

**Comment by yhmo** [comment]
**请下次提问时尽量使用英文。**
